### PR TITLE
Bind increment_version to python

### PR DESCRIPTION
**Pull request by albanD**
Should be convenient when writing python-only kernels (with triton) that don't have access to the C++ APIs.

### CODE REVIEW

File: torch/autograd/graph.py [hunk 1]
```diff
@@ -128,6 +129,21 @@ def __subclasshook__(cls, C):
...
+    """This function can be used to let autograd know that a given Tensor was modified
+    inplace to enable more accurate error checking within the autograd engine.
+
+    This is already done automatically by PyTorch functions and within custom Function
+    when mark_dirty() is called appropriately so you only need to call this explicitly
+    if you are doing inplace operation on the Tensor data in a way that Pytorch doesn't
+    know about. For example a custom kernel that reads the Tensor data_ptr and modifies
+    the memory inplace based on this pointer.
+
+    Note that incrementing the version counter multiple times for a single inplace operation
```
Comment (soulitzer) [inline torch/autograd/graph.py:142]: Curious when would someone want to do this?

**Review by soulitzer — APPROVED** [review]
LGTM, just small nits and a question

File: torch/autograd/graph.py [hunk 2]
```diff
@@ -128,6 +129,21 @@ def __subclasshook__(cls, C):
                 return True
         return NotImplemented
 
+def increment_version(tensor):
+    """This function can be used to let autograd know that a given Tensor was modified
+    inplace to enable more accurate error checking within the autograd engine.
+
+    This is already done automatically by PyTorch functions and within custom Function
+    when mark_dirty() is called appropriately so you only need to call this explicitly
```
[the remark this answers is not in the document]
Comment (albanD) [reply torch/autograd/graph.py:137]: I wrote this under the assumption that the user is doing the right thing. So this happens when this function is called. Not if it is called. Do you agree?

File: torch/autograd/graph.py [hunk 3]
```diff
@@ -128,6 +129,21 @@ def __subclasshook__(cls, C):
                 return True
         return NotImplemented
 
+def increment_version(tensor):
+    """This function can be used to let autograd know that a given Tensor was modified
```
[the remark this answers is not in the document]
Comment (albanD) [reply torch/autograd/graph.py:133]: The thing is that version counter is not really a public concept. That's why I left it as a longer version. WDYT?

[the remark this answers is not in the document]
Comment (soulitzer) [reply torch/autograd/graph.py:133]: Hmm, that makes sense. How about something like "Increments the version counter of a tensor to indicate that it has been modified inplace"

**Pull request closed without merging**